### PR TITLE
Add DSPv2 synthesis inference for qlf_k6n10f

### PR DESCRIPTION
**Pull request by lalit-quicklogic**
## Summary

Add DSPv2 (second-generation DSP block) Yosys-driven synthesis inference for the qlf_k6n10f architecture. This enables automatic mapping of RTL multiply, multiply-accumulate, and multiply-add patterns to `QL_DSPV2` cells via the `synth_quicklogic -dspv2` flag.

### Core synthesis passes
- **`ql-dsp.cc`** — New DSPv2 absorption pass: maps RTL multipliers to `QL_DSPV2` cells, absorbs input/output registers (`_REGIN`, `_REGOUT` variants), handles clock/reset adoption from behavioral RTL
- **`ql-dsp-macc.cc`** — Extended MACC inference to support DSPv2 multiply-accumulate patterns
- **`ql-dsp-simd.cc`** — Extended SIMD packing to support DSPv2 cell variants
- **`ql_dspv2_types.cc`** — Refactored `QL_DSPV2_TYPES` pass: analyzes port connectivity to assign typed wrappers (`QL_DSPV2_MULT`, `QL_DSPV2_MULTACC`, `QL_DSPV2_MULTADD`) with register variants
- **`synth_quicklogic.cc`** — Added `-dspv2` flag to `synth_quicklogic` command, wires up the DSPv2 flow (load sim/map/final_map files, run QL_DSPV2_TYPES pass)

### Library files
- **`dspv2_sim.v`** — DSPv2 simulation model (QL_DSPV2 blackbox)
- **`dspv2_map.v`** — DSPv2 techmap rules for inference
- **`dspv2_final_map.v`** — Final mapping to typed cell wrappers
- **`QL_DSPV2.v`** — Minor fix to blackbox definition
- Removed obsolete `dsp_final_map.v`

### Test suite (14 new tests)
- Multiply lowering, SIMD packing, MAC variants, register absorption
- Input register edge cases (ADFF, SDFF, B-reg, C-reg)
- Rejection tests (unsigned inputs, mixed resets, broken DFFRE fallback)
- Behavioral register absorption smoke tests
- Equivalence checking test

## Key design decisions
- DSPv2 is **signed-only** (max 32x18 signed multiply) — unsigned inputs are rejected
- Register absorption uses `_REGIN`/`_REGOUT` cell variants instead of external DFFRE wrapping
- Mixed-reset DFF absorption is rejected to prevent incorrect behavior
- The `-dspv2` flag is opt-in and does not affect existing DSPv1 flows

## Testing
- All 14 new DSPv2 unit tests pass
- 14 integration tests in Aurora2 pass (synthesis-only)
- Existing smoke tests unaffected

## Dependencies
- FOEDAG: quicklogic-corp/FOEDAG#166 (adds `-dspv2` option to compiler)
- device_data: quicklogic-corp/aurora-openfpga-plugins#237 (DSPv2 library files in device dirs)

### CODE REVIEW

File: ql-qlf-plugin/ql-dsp-macc.cc [hunk 1]
```diff
@@ -74,36 +75,71 @@ static void create_ql_macc_dsp(ql_dsp_macc_pm &pm)
     bool a_signed = st.mul->getParam(ID(A_SIGNED)).as_bool();
     bool b_signed = st.mul->getParam(ID(B_SIGNED)).as_bool();
 
+    // DSPv2 (QL_DSPV2) is signed-only. Reject unsigned multiplies and let them
+    // fall through to the mul2dsp+dspv2_map.v techmap (which inserts soft sign
+    // extension before the hard DSP).
+    if (dspv2 && (!a_signed || !b_signed)) {
```
Comment (tpagarani) [inline ql-qlf-plugin/ql-dsp-macc.cc:81]: DSPV2 supports unsigned multiplication. Why is there an assumption that it does not

File: ql-qlf-plugin/qlf_k6n10f/dsp_final_map.v [hunk 2]
```diff
@@ -494,208 +494,3 @@ generate
 endgenerate
 
 endmodule
```
Comment (tpagarani) [inline ql-qlf-plugin/qlf_k6n10f/dsp_final_map.v:496]: This code is required for DSPV2 IP Generation. Without this IP Generator flow will break It has to be updated to target the correction Functional modes based on new functional modes definition created by @saaramahmoudi

File: ql-qlf-plugin/ql-dsp-macc.cc [hunk 3]
```diff
@@ -213,29 +265,64 @@ static void create_ql_macc_dsp(ql_dsp_macc_pm &pm)
...
+        // dspv2_16x9x32_cfg_ports has neither unsigned_a/b_i (signed-only)
+        // nor the saturate/shift/round/reg_inputs/subtract ports of v1; those
+        // controls are cfg-parameters on the v2 wrapper. Tie unused cascade
+        // and second-operand inputs.
+        cell->setPort(RTLIL::escape_id("c_i"), RTLIL::SigSpec(RTLIL::S0, 9));
+        // Sync resets go to acc_reset_i (synchronous accumulator clear);
+        // async resets go to reset_i (already set above). When sync_reset
+        // is true, override reset_i back to S0 so the DSP's async reset
+        // path is not triggered.
+        if (sync_reset) {
```
Comment (tpagarani) [inline ql-qlf-plugin/ql-dsp-macc.cc:280]: Not quite clear why sync reset is being implemented using Accumulator reset

File: ql-qlf-plugin/ql_dspv2_types.cc [hunk 4]
```diff
@@ -423,82 +423,101 @@ struct QlDSPV2TypesPass : public Pass {
 					RTLIL::IdString("\\z"),
 					RTLIL::IdString("\\z_cout")
 				);
```
Comment (tpagarani) [inline ql-qlf-plugin/ql_dspv2_types.cc:425]: All this code from @navidjafarof has been removed inadvertently. Without this the registers won't be added at the inputs and outputs of DSP as required

File: ql-qlf-plugin/ql_dspv2_types.cc [hunk 5]
```diff
@@ -423,82 +423,101 @@ struct QlDSPV2TypesPass : public Pass {
...
+				// reset polarity, unconnected enable).
+				//
+				// C_REG is excluded: for MULT-class types the c port
+				// is dropped by the type rewrite, so C_REG is
+				// harmlessly ignored.  No external dffre is created.
+				if (!use_regin) {
+					bool need_ext_ab_reg =
+						(A1_REG || A2_REG || B1_REG || B2_REG ||
+						 A_REG || B_REG);
+					if (need_ext_ab_reg)
```
Comment (tpagarani) [inline ql-qlf-plugin/ql_dspv2_types.cc:480]: Auto generated code doesn't do register insertion anymore. It throws an error instead. The logic here does not seem to be correct

**Review by tpagarani — CHANGES_REQUESTED** [review]

**Comment by lalit-quicklogic** [comment]
closing this PR as created another one for this support
